### DISCUSSION TITLE
Sum of only positive numbers

### DISCUSSION THREAD

**Discussion by l-you** [doctrine/doctrine2#10029] [Support Questions] [answered]
I have large and complex query. One of columns is computed value that need sum of only positive numbers. Alternative in SQL is:
`SUM( IF( shop_pv.stockQuantity, shop_pv.stockQuantity, 0 ) ) > 0`.
In Doctrine there no `IF` function.  Problem can be resolved with subquery, but drawback is 600ms query execution instead 250ms.
Any idea on how to keep performance and do not use `leftJoin`'s that break query logic?

**Comment by SenseException** [comment, marked as answer]
`IF` sounds like you're using MySQL. If this question is about writing conditions in DQL, maybe this code snippet might help you to write your query:

`CASE WHEN f.name = 'foo' THEN 'its foo' ELSE 'NOT FOO' END AS qux`

This is only the condition part of an DQL.

```
array:1 [
  "qux" => "its foo"
]
array:1 [
  "qux" => "NOT FOO"
]
```
